### DISCUSSION TITLE
Accessing author metadata in shortcodes

### DISCUSSION THREAD

**Discussion by Nenuial** [quarto-dev/quarto-cli#2407] [Q&A] [answered]
I'm trying to access the author information in a shortcode and I can't find a way to access it. I think it has to do with the changes the author lua filter does to the metadata, but I might be mistaken.

My document has the following metadata :

```
---
date: 2022-09-01
author: John Doe
format: html
---

{{< doc-source >}}
```

I've tried dumping the `meta["author"]` variable but all I get is `WARNING: Unsupported type 'List' for key author` and `<empty table>`. I've also tried a number of combinations using `meta["author.something"]` including name, name.literal, etc.

Here's what my short shortcode code looks like for now:

```lua
return {
  ["doc-source"] = function(args, kwargs, meta)
    quarto.utils.dump(meta["author"])

    return nil
  end
}
```

Any pointers on how I could proceed to get the necessary information?

**Comment by dragonstyle** [comment, marked as answer]
The safest way should be to access `meta['authors']` which should be present with a normalized schema described here (since we're normalizing the author into the `authors` key):

https://quarto.org/docs/journals/authors.html

Using this rather than access the `author` key directly will give you the benefit of the author normalization that we do, ensuring that no matter how the user writes the author, you're able to access it consistently / predictably. 

Does this work?

```lua
return {
  ["doc-source"] = function(args, kwargs, meta)
    quarto.utils.dump(meta["authors"])

    return nil
  end
}
```

**Reply by Nenuial** [reply]
Thanks for the suggestion. I'm afraid this doesn't work either.

I get this output:
```
<empty table>
WARNING: Unsupported type 'List' for key authors
```

**Reply by dragonstyle** [reply]
You are completely correct- we were doing some normalization to the metadata for shortcodes that didn't properly handle lists. I've fixed this and the fix should be in the next pre-release (1.2.95) available here: https://quarto.org/docs/download/prerelease.html

Thanks for reporting this!

**Reply by Nenuial** [reply]
Works great! Thanks for the fix.
